pkg/client/client_test.go

```diff
@@ -0,0 +1,345 @@
+		},
+
+		"should return clusterID if clusterID is in the cache": {
+			clusterNameOrIDInput:                 clusterID,
+			clusterIdsCache:                      map[string]any{clusterID: struct{}{}},
+			clustersDisplayNameToIDCache:         map[string]any{clusterDN: clusterID},
+			fakeDynClient:                        dynamicfake.NewSimpleDynamicClient(scheme()),
+			expectedClusterIdsCache:              map[string]any{clusterID: struct{}{}},
+			expectedClustersDisplayNameToIDCache: map[string]any{clusterDN: clusterID},
+			expectedID:                           clusterID,
+		},
+
+		"should return clusterID if displayName is in the cache": {
+			clusterNameOrIDInput:                 clusterDN,
+			clusterIdsCache:                      map[string]any{clusterID: struct{}{}},
+			clustersDisplayNameToIDCache:         map[string]any{clusterDN: clusterID},
+			fakeDynClient:                        dynamicfake.NewSimpleDynamicClient(scheme()),
+			expectedClusterIdsCache:              map[string]any{clusterID: struct{}{}},
+			expectedClustersDisplayNameToIDCache: map[string]any{clusterDN: clusterID},
+			expectedID:                           clusterID,
+		},
+
+		"local": {
+			clusterNameOrIDInput:                 "local",
+			expectedClusterIdsCache:              map[string]any{},
+			expectedClustersDisplayNameToIDCache: map[string]any{},
+			expectedID:                           "local",
+		},
+
+		"cluster not found": {
+			clusterNameOrIDInput:                 clusterDN,
+			fakeDynClient:                        dynamicfake.NewSimpleDynamicClient(scheme(), newFakeCluster(clusterID, "another cluster")),
+			expectedClusterIdsCache:              map[string]any{clusterID: struct{}{}},
+			expectedClustersDisplayNameToIDCache: map[string]any{"another cluster": clusterID},
+			expectErr:                            "cluster 'my-display-name' not found",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			clusterIdsCache = sync.Map{}
+			if test.clusterIdsCache != nil {
+				for key, value := range test.clusterIdsCache {
+					clusterIdsCache.Store(key, value)
+				}
+			}
+			clustersDisplayNameToIDCache = sync.Map{}
+			if test.clustersDisplayNameToIDCache != nil {
+				for key, value := range test.clustersDisplayNameToIDCache {
+					clustersDisplayNameToIDCache.Store(key, value)
+				}
+			}
+
+			c := &Client{
+				DynClientCreator: func(inConfig *rest.Config) (dynamic.Interface, error) {
+					return test.fakeDynClient, nil
+				},
+			}
+
+			clusterID, err := c.getClusterId(fakeToken, fakeUrl, test.clusterNameOrIDInput)
+
+			if test.expectErr != "" {
+				require.ErrorContains(t, err, test.expectErr)
+			} else {
+				require.NoError(t, err)
+			}
+			assert.Equal(t, test.expectedID, clusterID)
+			assert.True(t, compareMapWithSyncMap(test.expectedClusterIdsCache, &clusterIdsCache))
+			assert.True(t, compareMapWithSyncMap(test.expectedClustersDisplayNameToIDCache, &clustersDisplayNameToIDCache))
```

Comment:
Maybe this should extract the map from the sync.Map and use `assert.Equal` to help clarify the differences?

Reply:
done